### ISSUE TITLE
"Be the Robot" plugin compatibility

### ISSUE THREAD

**Issue by naydef** [50DKP/FF2-Official#361]
There are servers, including own my own server, which have Be the Robot plugin(https://forums.alliedmods.net/showthread.php?t=193067). It's fun to use this plugin, but the boss can use it to change his model to robot or or his class model. This plugin has 3 natives:
```sourcemod
/**
 * Checks if a client is a robot.
 *
 * @param client			Index of the client to check.
 * @return					RobotStatus value of the client; RobotStatus_Human if they're a human, RobotStatus_Robot if they're a robot, RobotStatus_WantsToBeRobot if they want to be a robot, but can't for some reason.
 */
native RobotStatus:BeTheRobot_GetRobotStatus(client);

/**
 * Sets if a client should be robot or not.
 *
 * @param client			Index of the client to set.
 * @param toggle			True to make the client a robot, false to change them back to a human. Skip this argument to toggle instead.
 * @noreturn
 */
native BeTheRobot_SetRobot(client, bool:toggle = bool:2);

/**
 * Uses Be the Robot's "CheckTheRules()" native to check if a client should be allowed to be a robot or not.
 *
 * @param client			Index of the client to check.
 * @return					True if the client is allowed to be a robot (not dead, allowed class by server, etc.) false otherwise.
 */
native bool:BeTheRobot_CheckRules(client);
```
The code can be added to MakeBoss timer:
```sourcemod
if(BeTheRobot_GetRobotStatus(client))
{
     BeTheRobot_SetRobot(client, false);
}
```
or it can be added to Freak Fortress 2 compatibility subplugin for FF2v2.0. This reminds me why there isn't a forward, which is called when MakeBoss timer is called. It will be useful, when some subplugin want to for example give weapon, set attached particles, etc.

**Comment by 50Wliu** [comment]
Sure, this can be added to ff2_external_integration.

**Comment by naydef** [comment]
Yes, but the tiny problem is that ff2 can't detect if the player becomes robot. Maybe just will paste this code in some timer to check if the player is a robot and if he is, we will call BeTheRobot_SetRobot(client, false). Anyway, it seems that this plugin is not supported, maybe someone will take over this plugin and add forward. We will see.

**Comment by WildCard65** [comment]
AddCommandListener on the command and return Plugin_Handled or Plugin_Stop in the listener callback if caller is boss?

**Comment by naydef** [comment]
Yes. Don't use Plugin_Stop. Never!

**Comment by WildCard65** [comment]
Oh and why is that?

**Comment by naydef** [comment]
Originally Posted by Naydef

> 

26. Why in sound hook, Plugin_Stop is used instead of Plugin_Handled? I read that Plugin_Handled stop emitting sound, but Plugin_Stop stop emitting sound and stop the hook chain. What does it mean. Does it mean that it's like "for" loop and Plugin_Stop is like break for this loop?
> 

> 

Yes, it is like this (roughly):
Code:
```sourcemod
foreach (plugin) {
  call plugin callback();
  if (Plugin_Stop)
    break;
}
if (!Plugin_Handled && !Plugin_Stop) {
  call game functiion();
}
```
You should generally never use Plugin_Stop.

>

**Comment by Deathreus** [comment]
There's several places you have to use it though, you use it to stop a repeating timer, and you use it in sound hooks to stop the original sound from playing. You can also use it in a select few events and forwards to prevent them from firing.

**Comment by WildCard65** [comment]
@naydef I was talking about AddCommandListener, not a sound hook
